### ISSUE TITLE
Performance on JUWELS-Booster

### ISSUE THREAD

**Issue by goord** [microhh/rte-rrtmgp-cpp#6]
I have completed my tests on juwels-booster, which has 2 x 24C AMD-EPYC Rome cpu's and 4 x NVIDIA A100 per node. I see virtually no speedup yet between CPU/GPU versions
| case | config  | longwave time [ms] | shortwave time [ms] |
|:-:|:-:|:-:|---|
| allsky | gcc | 251 | 257 |
| allsky | gcc+cuda | 255 | 258 |
| rfmip | gcc | 316 | 255 |
| rfmip | gcc+cuda | 323 | 260 |

For the CUDA runs I used the cuda branch.

**Comment by Chiil** [comment]
Can you also test the `rcemip` testcase and increase the work load to by setting `n_col` to `64**2`? The other cases are relatively cheap and are mainly used for correctness checking.

**Comment by goord** [comment]
Yes for some reason I couldn't get the input file for the rcemip case... will try once more.

**Comment by Chiil** [comment]
Send me a message on Slack if I need to clarify something, maybe the `make_links.sh` misses a step.

**Comment by isazi** [comment]
I managed to run all three tests on the DAS.

**Comment by goord** [comment]
Ah I first needed to run ``test_rcemip_input.py``. Ok, here is the updated table:

| case | config  | longwave time [ms] | shortwave time [ms] |
|:-:|:-:|:-:|---|
| allsky | gcc | 251 | 257 |
| allsky | gcc+cuda | 255 | 258 |
| rfmip | gcc | 316 | 255 |
| rfmip | gcc+cuda | 323 | 260 |
| rcemip | gcc | 38418 | 35908 |
| rcemip | gcc+cuda | 37622 | 35871 |

**Comment by isazi** [comment]
Would it make sense if I also provide the same table for DAS-5? We have a node with an A100 (but older and slower Xeon CPUs).

**Comment by goord** [comment]
You can tell me if those numbers are in line with mine Alessio.

**Comment by goord** [comment]
Fixed my latest table which was transposed

**Comment by Chiil** [comment]
I did a quick benchmark on an AWS P3 V100 instance:

| case |  config | longwave (ms) | shortwave (ms) |
| --- | --- | --- | --- |
| rcemip | gcc+cuda | 3117 | 2331 |
| rcemip | gcc | 14365 | 11362 |

**Comment by goord** [comment]
After using fortran compiler flags (!) I get following figures:

| case |  config | longwave (ms) | shortwave (ms) |
| --- | --- | --- | --- |
| rcemip | gcc+cuda | 3688 | 3319 |
| rcemip | gcc | 8084 | 6982 |

So A100's (at least the ones in juwels-booster) appear somewhat slower than the V100 card. This may indicate the code needs some re-tuning...

**Comment by Chiil** [comment]
We have not done any tuning yet. We have rushed a little in getting a reference implementation ready that gives identical results with the CPU version, but there is probably still a lot to gain in kernel tuning.

**Comment by goord** [comment]
Some ad-hoc tuning of the kernel block sizes reduces the timings to 3051 ms (lw) and 2669 ms (sw) so yes, there is definitely headroom for speedup by tuning on A100s.

**Comment by goord** [comment]
Further manual tuning established 

| case |  config | longwave (ms) | shortwave (ms) |
| --- | --- | --- | --- |
| rcemip | gcc+cuda | 1439 | 1174 |
